src/tests/base/wsteth15.test.ts

```diff
@@ -7,94 +6,69 @@ import {
-      chainId,
-      isMinting: true,
-      inputToken: usdc,
-      outputToken: indexToken,
-      indexTokenAmount: wei('1').toString(),
-      inputTokenAmount: wei('2100', 6).toString(),
-      slippage: 0.5,
-    })
-    const whale = '0x621e7c767004266c8109e83143ab0da521b650d6'
-    await transferFromWhale(
-      whale,
-      factory.getSigner().address,
-      wei('10000', quote.inputToken.decimals),
-      quote.inputToken.address,
-      factory.getProvider(),
-    )
-    await factory.executeTx()
-  })
+  for (const {
+    setAmount,
+    inputAmount,
+    inputToken: inputTokenSymbol,
+  } of testCases) {
+    describe(`SetAmount: ${setAmount} - inputAmount ${inputAmount}`, () => {
+      let quote: any
 
-  test('can mint with WETH', async () => {
-    const quote = await factory.fetchQuote({
-      chainId,
-      isMinting: true,
-      inputToken: weth,
-      outputToken: indexToken,
-      indexTokenAmount: wei('1').toString(),
-      inputTokenAmount: wei('1.1').toString(),
-      slippage: 0.5,
-    })
-    await wrapETH(
-      quote.inputAmount.mul(BigNumber.from(2)),
-      factory.getSigner(),
-      chainId,
-    )
-    await factory.executeTx()
-  })
+      beforeAll(async () => {
+        factory.resetFork(chainId)
 
-  test('can redeem to ETH', async () => {
-    await factory.fetchQuote({
-      chainId,
-      isMinting: false,
-      inputToken: indexToken,
-      outputToken: ETH,
-      indexTokenAmount: wei('1').toString(),
-      inputTokenAmount: wei('1').toString(),
-      slippage: 0.5,
-    })
-    await factory.executeTx()
-  })
+        const inputToken =
+          inputTokenSymbol === 'ETH'
+            ? ETH
+            : getTokenByChainAndSymbol(chainId, inputTokenSymbol)
+
+        quote = await factory.fetchQuote({
+          chainId,
+          isMinting: true,
+          inputToken: inputToken!,
+          outputToken: indexToken,
+          indexTokenAmount: wei(setAmount).toString(),
+          inputTokenAmount: wei(inputAmount, inputToken!.decimals).toString(),
+          slippage: 0.5,
+        })
+      })
+
+      test('can obtain quote', () => {
+        expect(quote).not.toBeNull()
+      })
+
+      test('can mint', async () => {
+        if (!quote) {
+          throw new Error("Can't mint without quote")
+        }
+
+        if (inputTokenSymbol !== 'ETH') {
+          await transferFromWhale(
+            whale,
+            factory.getSigner().address,
+            quote.inputOutputAmount,
+            quote.inputToken.address,
+            factory.getProvider(),
+          )
+        }
 
-  test('can redeem to USDC', async () => {
-    await factory.fetchQuote({
-      chainId,
-      isMinting: false,
-      inputToken: indexToken,
-      outputToken: usdc,
-      indexTokenAmount: wei('1').toString(),
-      inputTokenAmount: wei('1').toString(),
-      slippage: 0.5,
+        await factory.executeTx()
+      })
     })
-    await factory.executeTx()
-  })
+  }
```

Comment:
I think I got this to work like you were thinking @ckoopmann. Let me know. Probably best known viewed in `View file` now with all these changes lol

Reply:
If so, we could adapt other tests accordingly in subsequent PRs.